lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -8750,17 +8764,21 @@ void ADContext::foldDifferentiableFunctionExtraction(
 
 bool ADContext::processDifferentiableFunctionInst(
     DifferentiableFunctionInst *dfi) {
+  PrettyStackTraceSILNode dfiTrace("canonicalizing `differentiable_function`",
+                                   cast<SILInstruction>(dfi));
+  PrettyStackTraceSILFunction fnTrace("...in", dfi->getFunction());
```

Comment:
Note: this pattern of `PrettyStackTraceSILNode` followed by `PrettyStackTraceSILFunction` is used because `PrettyStackTraceSILNode` by itself does not print any info about the containing `SILFunction`.

I copied this technique from [elsewhere in the codebase](https://github.com/apple/swift/blob/27dfbd1936a565b8990a00be49fdc09026c991b7/lib/AST/ASTContext.cpp#L1640). It's less intrusive than adding a flag to `PrettyStackTraceSILNode` for printing SIL function info.

Example stack trace:
```
Stack dump:
...
3.	While canonicalizing `differentiable_function` SIL node   %18 = differentiable_function [parameters 0] %17 : $@callee_guaranteed (Float, Int) -> Float // users: %22, %19
4.	While ...in SIL function "@main".
```